### PR TITLE
Tests: Do validation for integration NWB files as well

### LINKED ISSUE #928: Created NWB files during CI don't validate
See #927.

The errors are

```
[...]
======================================================================
2019-05-13 18:26:33,350 - INFO - Validating file test_PlaneSegmentation.nwb
Error: ImagingPlane/manifold (general/optophysiology/ophys integration tests/manifold): incorrect shape - expected '[[None, None, 3], [None, None, None, 3]]', got'(5,)'
======================================================================
2019-05-13 18:26:33,527 - INFO - Validating file test_RoiResponseSeries.nwb
Error: ImagingPlane/manifold (general/optophysiology/ophys integration tests/manifold): incorrect shape - expected '[[None, None, 3], [None, None, None, 3]]', got'(5,)'
======================================================================
[...]
======================================================================
2019-05-13 18:26:37,396 - INFO - Validating file test_EventWaveform.nwb
Error: SpikeEventSeries/data (acquisition/EventWaveform/test_sES/data): incorrect shape - expected '[[None, None], [None, None, None]]', got'(10,)'
======================================================================
[...]
2019-05-13 18:26:38,573 - INFO - Ran 363 tests - FAILED (errors=3,failures=3)
```

### PR DESCRIPTION
**Pull request by t-b**
## Motivation

We need to validate more files.

## How to test the behavior?
```
View CI passing
```

Close #928.

### CODE REVIEW

**Comment by t-b** [comment]
@rly Any idea how to fix the validation errors?

**Comment by t-b** [comment]
@rly Thanks! Your changes did the trick.

File: test.py [hunk 1]
```diff
@@ -208,11 +208,13 @@ def main():
     # Run example tests
     if flags['example'] in args.suites:
         run_example_tests()
-        validate_example_nwbs()
+        validate_nwbs()
 
     # Run integration tests
     if flags['integration'] in args.suites:
+        os.environ['CLEAN_NWB'] = 'False'
```
Comment (rly) [inline test.py]: Is there an easy way to remove all of the created files after validation is completed?

Comment (rly) [reply test.py]: Perhaps add a parameter to `validate_nwbs` to set `delete_files` = True/False and then within the loop over files, check `delete_files` and delete the file if true.

Comment (t-b) [reply test.py]: @rly Or maybe the validation should not happen here but in the integration tests itself and we can remove the `CLEAN_NWB` fiddling here?

File: tests/integration/test_io.py [hunk 2]
```diff
@@ -237,24 +237,29 @@ def test_write_cache_spec(self):
 
 class TestAppend(unittest.TestCase):
 
-    def test_append(self):
+    def setUp(self):
```
Comment (rly) [inline tests/integration/test_io.py:240]: These changes just clean up this unit test and reform it to be like other tests. It also makes sure that the created file gets deleted.

File: tests/integration/ui_write/base.py [hunk 3]
```diff
@@ -185,6 +185,16 @@ def test_roundtrip(self):
...
+        self.validate()
+
+    def validate(self):
+        # validate created file
+        if os.path.exists(self.filename):
+            with NWBHDF5IO(self.filename, mode='r') as io:
+                errors = pynwb_validate(io)
+                if errors:
+                    for err in errors:
+                        raise Exception(err)
```
Comment (rly) [inline tests/integration/ui_write/base.py:198]: This is the core change -- all roundtrip tests will be validated before the file gets deleted. Validation errors become exceptions.

File: tests/integration/ui_write/test_modular_storage.py [hunk 4]
```diff
@@ -78,6 +79,23 @@ def roundtripContainer(self):
...
+                            raise Exception(err)
+
+                # necessary to remove all references to the file and garbage
+                # collect on windows in order to be able to truncate/overwrite
+                # the file later. see pynwb GH issue #975
+                if os.name == 'nt':
+                    del io
+                    gc.collect()
+
     def getContainer(self, nwbfile):
```
Comment (rly) [inline tests/integration/ui_write/test_modular_storage.py:99]: This test creates two linked files and so validation needed to be handled specially.

**Review by ajtritt — APPROVED** [review]

File: tests/integration/ui_write/test_ecephys.py [hunk 5]
```diff
@@ -320,7 +320,7 @@ def setUpContainer(self):
         TestElectricalSeriesIO.make_electrode_table(self)
         region = DynamicTableRegion('electrodes', [0, 2], 'the first and third electrodes', self.table)
         sES = SpikeEventSeries(
-            'test_sES', list(range(10)), list(range(10)), region)
+            'test_sES', ((1, 1, 1), (2, 2, 2)), list(range(2)), region)
         ew = EventWaveform(sES)
```
Comment (rly) [inline tests/integration/ui_write/test_ecephys.py:324]: The inputs to SpikeEventSeries were the wrong shape here.

File: tests/integration/ui_write/test_ophys.py [hunk 6]
```diff
@@ -182,12 +182,14 @@ def buildPlaneSegmentation(self):
...
                                           self.optical_channel,
-                                          'imaging plane description',
+                                          'a fake ImagingPlane',
                                           self.device,
-                                          600., 300., 'GFP', 'somewhere in the brain',
-                                          (1., 2., 1., 2., 3.), 4.0, 'manifold unit', 'A frame to refer to')
+                                          600., 200., 'GFP', 'somewhere in the brain',
+                                          (((1., 2., 3.), (4., 5., 6.)),),
+                                          2., 'a unit',
+                                          reference_frame='unknown')
```
Comment (rly) [inline tests/integration/ui_write/test_ophys.py:193]: The parameters of ImagingPlane were the wrong shape here.

**Comment by t-b** [comment]
@rly Thanks for taking over!!

**Pull request merged**